components/MapFilters.vue

```diff
@@ -0,0 +1,249 @@
+             @click="toggleFilters()">
+          <polyline points="18 15 12 9 6 15">
+          </polyline>
+        </svg>
+      </span>
+    </div>
+    <!-- Filters pop-down -->
+    <div id="map-filters" :class="filters ? 'block': 'hidden'"
+         class="pl-5 pr-5 pt-3 pb-3 text-xs">
+      <div>
+        <span v-for="(tile, index) in tiles">
+          <input type="radio" name="tile"
+                 :checked="tile.name === currentTileName"
+                 @change="filterTiles(tile)"/>
+          &nbsp;{{tile.name}}&nbsp;
+        </span>
+      </div>
+      <br>
+      <div class="flex flex-col">
+        <div>
+          <span v-for="(layer, index) in filterLayers">
+            <input type="radio" name="layer"
+                   :checked="index === currentLayer"
+                   @change="toggleFilterLayers(index)"/>
+            &nbsp;{{layer.label}}&nbsp;
+          </span>
+        </div>
+        <div class="pt-2">
+          <span class="flex items-center float-left"
+                v-for="(item, filterType) in filterLayers[currentLayer].filters">
+            <input type="checkbox" v-model="item.active"
+                   @change="filterMarkers(filterType)"/>
+            &nbsp;
+            <img :src="item.iconUrl" width="12" height="20"/>
+            &nbsp;{{item.label}}&nbsp;
+          </span>
+        </div>
+      </div>
+      <br>
+    </div>
+  </div>
+</template>
+
+<script>
+ export default {
+   props: {
+     startingMarkers: {
+       type: Array,
+       required: true
+     },
+     filterLayers: {
+       type: Array,
+       required: true
+     },
+     isLoading: {
+       type: Boolean,
+       required: true
+     }
+   },
+   data () {
+     return {
+       filters: false,
+       noFiltersOn: 0,
+       markers: [],
+       originalMarkers: [],
+       currentTileName : 'Modern Map',
+       tiles: [{name: 'Modern Map', url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', active: true},
+               {name: 'Historic Map', url: 'https://api.maptiler.com/tiles/uk-osgb1919/{z}/{x}/{y}.jpg?key=cKVGc9eOyhb8VH5AxCtw', active : false}],
+       currentLayer: 0, // Index in filterLayers corresponding to current layer.
+     }
+   },
+   watch: {
+     startingMarkers: {
+       handler(newValue, oldValue) {
+         this.originalMarkers = JSON.parse(JSON.stringify(newValue));
+         this.markers = JSON.parse(JSON.stringify(newValue));
+       }
+     }
+   },
+   methods: {
+     getMarkerType : function (marker) {
+       // Returns the marker type. If it detects that
+       // one witch has a different type to the others,
+       // returns mixed straight away. Otherwise returns
+       // markerType, which will be common to all witches.
+
+       let markerType = '';
+       let witches = marker.witches;
+       for (let i = 0, len = witches.length; i < len; i++) {
+         let witch = witches[i];
+         let witchType = witch[this.currentFilterProperty];
+         if (i === 0) {
```

Comment:
Same as for statement. Control structures should have an empty line above an below.

So 

```
let witchType = witch[this.currentFilterProperty];
if (i === 0) {
  markerType = witchType;
} else if (witchType !== markerType) {
  return 'mixed';
}
```

Should be 

```
let witchType = witch[this.currentFilterProperty];

if (i === 0) {
  markerType = witchType;
} else if (witchType !== markerType) {
  return 'mixed';
}
```